packages/kbn-apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/apm_pipeline.ts

```diff
@@ -48,6 +53,7 @@ export function apmPipeline(logger: Logger, version: string, includeSerializatio
       createBreakdownMetricsAggregator('30s'),
       getApmServerMetadataTransform(version),
       getRoutingTransform(),
+      ...pipelines,
```

Comment:
I see your use case, you want to inject a new pipeline. You can do this with exisitng logic without touching synthttrace client like here

https://github.com/elastic/kibana/blob/bb08db94e0207e4bb5785a287abb21c5a34762c8/x-pack/test/apm_api_integration/tests/time_range_metadata/generate_data.ts#L65-L76